src/vtquery.cpp

```diff
@@ -28,9 +30,12 @@ const char* getGeomTypeString(int enumVal) {
     return GeomTypeStrings[enumVal];
 }
 
+using materialized_prop_type = std::pair<std::string, mapbox::feature::value>;
```

Comment:
Yes, that comment is the intended documentation for this. Basically I think we need to copy the data to safely use it outside of the scope of the `std::string` that represents the original decompressed tile buffer. I used `materialized` as a fancy way to describe this copy. Other terms could be `standalone_prop_type` or `persistent_prop_type` or something.

Reply:
[nitpick] Since T const& instead of const T& is the style used throughout this code file, it might make sense to use this style for pointers as well. So instead of `const char*` and `static const char*` you would use `char const*` and `static char const*` respectively. It looks weird to me but it's consistent!

Reply:
@alliecrevier not quite sure which chunk of code this comment refers to. Perhaps you meant to place against another line?

Reply:
Ah must be https://github.com/mapbox/vtquery/blob/f98b2b65b8374ef01bacf8bca96867d088d46ac4/src/vtquery.cpp#L28-L29 (which I could not see in the github comment display since it only shows the first couple lines above the comment).

Reply:
I'm going to not touch this code right now since the differences are a little above my knowledge to feel comfortable with. Let's discuss in a ticket: https://github.com/mapbox/vtquery/issues/96